documentation-site/components/example.js

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+/* global fetch process */
+
+import * as React from 'react';
+import CodeSandboxer from 'react-codesandboxer';
+import {CopyToClipboard} from 'react-copy-to-clipboard';
+
+import {Button, KIND} from 'baseui/button';
+import {Card} from 'baseui/card';
+import {Block} from 'baseui/block';
+import Check from 'baseui/icon/check';
+import {StyledLink} from 'baseui/link';
+import {styled} from 'baseui/styles';
+
+const Link = styled(StyledLink, {cursor: 'pointer'});
+
+const index = `
+import React from "react";
+import ReactDOM from "react-dom";
+
+import { Provider as StyletronProvider } from "styletron-react";
+import { Client as Styletron } from "styletron-engine-atomic";
+
+import Example from "./example";
+
+const engine = new Styletron();
+
+function App() {
+  return <Example />;
+}
+
+const rootElement = document.getElementById("root");
+ReactDOM.render(
+  <StyletronProvider value={engine}>
+    <App />
+  </StyletronProvider>,
+  rootElement
+);
+`;
+
+function Source(props: {children: ?React.Node}) {
+  if (!props.children || typeof props.children !== 'string') return null;
+
+  return (
+    <Block
+      padding="scale800"
+      overrides={{Block: {style: {fontFamily: 'courier'}}}}
+    >
+      {props.children
+        .trim()
+        .split('\n')
+        .map((line, index) => (
+          <Block key={index}>{line ? line : <br />}</Block>
+        ))}
+    </Block>
+  );
+}
+
+type PropsT = {
+  children: React.Node,
+  path: string, // required to fetch the uncompiled source code
+  title: string,
+};
+
+type StateT = {
+  isCopied: boolean,
+  source: ?string,
+};
+
+class Example extends React.Component<PropsT, StateT> {
+  state = {
+    isCopied: false,
+    source: null,
+  };
+
+  async componentDidMount() {
+    const sourcePath = `${String(process.env.STATIC_ROOT)}${this.props.path}`;
+    const res = await fetch(sourcePath);
+    const source = await res.text();
+    this.setState({source});
+  }
+
+  handleCopy = () => {
+    this.setState({isCopied: true}, () =>
+      setTimeout(() => this.setState({isCopied: false}), 5000),
```

Comment:
why do we need the setTimeout here?

Reply:
My thought is that a visitor would like some feedback to indicate the code was successfully copied. To do so, I replace the `copy to clipboard` text with `copied to clipboard`. Following up from that, I'd like to reset the text back to original message to communicate that it is still able to be copied

Reply:
ahhh, makes perfect sense - missed that!